starknet/rust/src/lib.rs

```diff
@@ -1,34 +1,61 @@
-use std::ffi::{c_char, CStr, CString};
 use cairo_lang_starknet_classes::casm_contract_class::CasmContractClass;
+use std::ffi::{c_char, CStr, CString};
 
 #[no_mangle]
-pub extern "C" fn compileSierraToCasm(sierra_json: *const c_char) -> *mut c_char {
-    let sierra_json_str = unsafe { CStr::from_ptr(sierra_json) }.to_str().unwrap();
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
+pub extern "C" fn compileSierraToCasm(sierra_json: *const c_char, result: *mut *mut c_char) -> u8 {
```

Comment:
why `result` type is *mut *mut ?

Reply:
Rust doesn't allow you to directly return a dynamically allocated string because Rust's memory management system would deallocate the string when the function returns.